src/socket_base.cpp

```diff
@@ -1289,6 +1295,7 @@ int zmq::socket_base_t::close ()
     //  process.
     send_reap (this);
 
+    EXIT_MUTEX ();
```

Comment:
Does the EXIT_MUTEX need to be after the send_reap?

The problem is that it causes read-after-free, as shown by Valgrind tests now failing. I guess the mutex object is freed before it returns.

Moving the mutex release before the send_reap makes the tests happy. Is it the right thing to do?